src/app/models/coffeeShop.model.ts

```diff
@@ -0,0 +1,9 @@
+export interface CoffeeShopModel {
+  id: number,
+  created_at: Date,
+  updated_at: Date,
+  name: string,
+  x: string,
+  y: string,
+  distanceFromUser: number;
+}
```

Comment:
I see the model as the place where we define how our model should look. `x` and `y` would ideally be `floats` or `numbers`, not strings, and it does look like it required you to do some forced conversions to `Number` in `app.component.ts`. Why couldn't they be declared as  a more useful type in here to avoid all that?

Reply:
I declared them as strings because this would be returned by the api. I presumed the api would return only strings, and didn't want to risk it, prefering to convert it to Number when I needed it.

Reply:
I've seen a pattern where there is a middle layer between the `HttpRequest` and the `Model` for transforming the values from the API into the values expected by the Model, something like (`Request >> Deserializer >> Model`). The layer would be a serializer/deserializer. This would be really helpful in this case.

Reply:
Yes, I am familiar with that pattern(i think). In which in the model you create a class that implements the model, and create a method within it that parses the ApiResponse. But however, recently I found out that it's not so performant, and you're better off using interfaces only. Correct me if I didn't understand.

Reply:
I cannot comment on the performance, but it is up to you what you think is more important. I personally would sacrifice a negligible amount of performance if the client code ends up cleaner and easier to understand. I believe the bottleneck in present day apps is with the rendering.